mlrun/runtimes/serving.py

```diff
@@ -322,11 +322,17 @@ def set_tracking(
 
         # Applying model monitoring configurations
         self.spec.track_models = True
-        self.spec.tracking_policy = model_monitoring.TrackingPolicy()
+        self.spec.tracking_policy = ""
         if tracking_policy:
-            self.spec.tracking_policy = self.spec.tracking_policy.from_dict(
-                tracking_policy
-            )
+            if isinstance(tracking_policy, dict):
+                # Convert tracking policy dictionary into `model_monitoring.TrackingPolicy` object
+                self.spec.tracking_policy = model_monitoring.TrackingPolicy().from_dict(
```

Comment:
```suggestion
                self.spec.tracking_policy = model_monitoring.TrackingPolicy.from_dict(
```

`from_dict` is a class-method, so can be invoked directly on the class. Your usage created an empty `TrackingPolicy` instance, and then used it to invoke `to_dict` and create another instance. It works, but is not the way to do it.